### PR TITLE
Updated codeowners

### PR DESCRIPTION
**Pull request by JesperIRL**
The codeowners list was updated with the latest new and removed directories. Also added links to directories.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace

### Reviewers
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**) ⚠️ Review applies to [0c3f4fe5](https://git.openjdk.org/guide/pull/120/files/0c3f4fe5164183721cc15edff4ea5397154f55cf)
 * [Stefan Karlsson](https://openjdk.org/census#stefank) (@stefank - no project role) ⚠️ Review applies to [0c3f4fe5](https://git.openjdk.org/guide/pull/120/files/0c3f4fe5164183721cc15edff4ea5397154f55cf)
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - Author) ⚠️ Review applies to [0c3f4fe5](https://git.openjdk.org/guide/pull/120/files/0c3f4fe5164183721cc15edff4ea5397154f55cf)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/guide.git pull/120/head:pull/120` \
`$ git checkout pull/120`

Update a local copy of the PR: \
`$ git checkout pull/120` \
`$ git pull https://git.openjdk.org/guide.git pull/120/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 120`

View PR using the GUI difftool: \
`$ git pr show -t 120`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/guide/pull/120.diff">https://git.openjdk.org/guide/pull/120.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/guide/pull/120#issuecomment-1954757175)

### CODE REVIEW

**Review by irisclark — APPROVED** [review]
Looks reasonable.  I'm not too familiar with the VM source layout, so I'd recommend somebody with that expertise take a look at this too, just in case.

**Comment by RogerRiggs** [comment]
Is this documenting the Skara bots mapping of files/directories to mailing lists?

File: src/guide/code-owners.md [hunk 1]
```diff
@@ -27,182 +26,189 @@ This list is intended to make it easier to identify which email list to include
...
+    * [crypto](https://github.com/openjdk/jdk/tree/master/src/java.base/share/classes/javax/crypto) - Security
+    * [[linux](https://github.com/openjdk/jdk/tree/master/java.base/linux/classes/jdk/internal), [macosx](https://github.com/openjdk/jdk/tree/master/java.base/macosx/classes/jdk/internal), [share](https://github.com/openjdk/jdk/tree/master/java.base/share/classes/jdk/internal), [unix](https://github.com/openjdk/jdk/tree/master/java.base/unix/classes/jdk/internal), [windows](https://github.com/openjdk/jdk/tree/master/java.base/windows/classes/jdk/internal)]/internal
+      * [access](https://github.com/openjdk/jdk/tree/master/src/java.base/share/classes/jdk/internal/access) - Core Libs, Security
+      * [event](https://github.com/openjdk/jdk/tree/master/src/java.base/share/classes/jdk/internal/event) - JFR
+      * [foreign](https://github.com/openjdk/jdk/tree/master/src/java.base/share/classes/jdk/internal/foreign) - Core Libs
+      * [icu](https://github.com/openjdk/jdk/tree/master/src/java.base/share/classes/jdk/internal/icu) - Core Libs
+      * [io](https://github.com/openjdk/jdk/tree/master/src/java.base/share/classes/jdk/internal/io) - NIO
```
Comment (AlanBateman) [inline src/guide/code-owners.md:73]: This should be Core Libs, not NIO.

File: src/guide/code-owners.md [hunk 2]
```diff
@@ -27,182 +26,189 @@ This list is intended to make it easier to identify which email list to include
...
+      * [javac](https://github.com/openjdk/jdk/tree/master/src/java.base/share/classes/jdk/internal/javac) - Java Language (javac)
+      * [jimage](https://github.com/openjdk/jdk/tree/master/src/java.base/share/classes/jdk/internal/jimage) - Client Libs
```
Comment (AlanBateman) [inline src/guide/code-owners.md:75]: This should Core Libs, not Client Libs.

File: src/guide/code-owners.md [hunk 3]
```diff
@@ -27,182 +26,189 @@ This list is intended to make it easier to identify which email list to include
...
+      * [jmod](https://github.com/openjdk/jdk/tree/master/src/java.base/share/classes/jdk/internal/jmod) - HotSpot
```
Comment (AlanBateman) [inline src/guide/code-owners.md:76]: This should be Core Libs.

File: src/guide/code-owners.md [hunk 4]
```diff
@@ -27,182 +26,189 @@ This list is intended to make it easier to identify which email list to include
...
+      * [ref](https://github.com/openjdk/jdk/tree/master/src/java.base/share/classes/jdk/internal/ref) - Core Libs, GC
+      * [reflect](https://github.com/openjdk/jdk/tree/master/src/java.base/share/classes/jdk/internal/reflect) - Core Libs
+      * [util/random](https://github.com/openjdk/jdk/tree/master/src/java.base/share/classes/jdk/internal/util/random) - Core Libs
+      * [util/regex](https://github.com/openjdk/jdk/tree/master/src/java.base/share/classes/jdk/internal/util/regex) - Core Libs
+      * [util/xml](https://github.com/openjdk/jdk/tree/master/src/java.base/share/classes/jdk/internal/util/xml) - Core Libs
+      * [vm](https://github.com/openjdk/jdk/tree/master/src/java.base/share/classes/jdk/internal/vm) - HotSpot
+    * [invoke](https://github.com/openjdk/jdk/tree/master/src/java.base/share/classes/sun/invoke) - Core Libs
+    * [[share](https://github.com/openjdk/jdk/tree/master/java.base/share/classes/java/io), [sun](https://github.com/openjdk/jdk/tree/master/java.base/windows/classes/sun/io), [unix](https://github.com/openjdk/jdk/tree/master/java.base/unix/classes/java/io)]/io - NIO
```
Comment (AlanBateman) [inline src/guide/code-owners.md:94]: This should be Core Libs.

File: src/guide/code-owners.md [hunk 5]
```diff
@@ -27,182 +26,189 @@ This list is intended to make it easier to identify which email list to include
...
+    * [[macosx](https://github.com/openjdk/jdk/tree/master/java.base/macosx/classes/java), [share](https://github.com/openjdk/jdk/tree/master/java.base/share/classes/java), [unix](https://github.com/openjdk/jdk/tree/master/java.base/unix/classes/java), [windows](https://github.com/openjdk/jdk/tree/master/java.base/windows/classes/java)]/java
+      * [[share](https://github.com/openjdk/jdk/tree/master/java.base/share/classes/java/lang), [unix](https://github.com/openjdk/jdk/tree/master/java.base/unix/classes/java/lang), [windows](https://github.com/openjdk/jdk/tree/master/java.base/windows/classes/java/lang)]/lang - Core Libs
+      * [math](https://github.com/openjdk/jdk/tree/master/src/java.base/share/classes/java/math) - Core Libs
+      * [time](https://github.com/openjdk/jdk/tree/master/src/java.base/share/classes/java/time) - Core Libs
+    * [launcher](https://github.com/openjdk/jdk/tree/master/src/java.base/share/classes/sun/launcher) - Tools
```
Comment (AlanBateman) [inline src/guide/code-owners.md:99]: That may be where the historical ownership was but any changes to the launcher tend to be discussed/reviews on core-libs so probably better to change this owner.

File: src/guide/code-owners.md [hunk 6]
```diff
@@ -27,182 +26,189 @@ This list is intended to make it easier to identify which email list to include
...
+    * [[share](https://github.com/openjdk/jdk/tree/master/java.base/share/native/libjimage), [unix](https://github.com/openjdk/jdk/tree/master/java.base/unix/native/libjimage), [windows](https://github.com/openjdk/jdk/tree/master/java.base/windows/native/libjimage)]/libjimage - Core Libs
+    * [[aix](https://github.com/openjdk/jdk/tree/master/java.base/aix/native/libjli), [macosx](https://github.com/openjdk/jdk/tree/master/java.base/macosx/native/libjli), [share](https://github.com/openjdk/jdk/tree/master/java.base/share/native/libjli), [unix](https://github.com/openjdk/jdk/tree/master/java.base/unix/native/libjli), [windows](https://github.com/openjdk/jdk/tree/master/java.base/windows/native/libjli)]/libjli - Tools
```
Comment (AlanBateman) [inline src/guide/code-owners.md:122]: "jli" = Java launcher infrastructure so should have same owner as java launcher.

**Review by AlanBateman — APPROVED** [review]
Looks fine, a few suggestions included.

**Comment by dholmes-ora** [comment]
With the addition of the links it is not easy to see the actual changes to the mapping, so hard to check without re-checking the entire list.

File: src/guide/code-owners.md (hunk 1 above)
Comment (JesperIRL) [reply src/guide/code-owners.md:73]: Fixed.

File: src/guide/code-owners.md (hunk 2 above)
Comment (JesperIRL) [reply src/guide/code-owners.md:75]: Fixed.

File: src/guide/code-owners.md (hunk 3 above)
Comment (JesperIRL) [reply src/guide/code-owners.md:76]: Fixed.

File: src/guide/code-owners.md (hunk 4 above)
Comment (JesperIRL) [reply src/guide/code-owners.md:94]: Fixed.

File: src/guide/code-owners.md (hunk 5 above)
Comment (JesperIRL) [reply src/guide/code-owners.md:99]: Added Core-Libs.

File: src/guide/code-owners.md (hunk 6 above)
Comment (JesperIRL) [reply src/guide/code-owners.md:122]: Fixed.

**Pull request closed without merging**